manager/usermanager/createUser.go

```diff
@@ -26,11 +28,28 @@ func CreateUser(userStore *store.UserStore, user *models.UserCredentials) (*mode
 	}
 
 	password := string(hashedPassword)
-	user.Password = &password
 	uid := uuid.Must(uuid.NewRandom()).String()
-	user.UID = &uid
-	err = userStore.Set(user)
-	return user.GetPublicInfo(), err
+	var role models.Role
+	if user.GetRole() != "" {
+		role = user.Role
+	} else {
+		role = models.RoleUser
+	}
+
+	newUser := &models.UserCredentials{
```

Comment:
Why this change?

Reply:
This is to limit the fields that the new user is allowed to give as this function is being called from here too https://github.com/mayadata-io/kubera-auth/pull/45/files#diff-104a11712684e520f0affa95e6a053faeb713306a10f69ee99b9729cc9dd1996R272